node/src/Pos/Wallet/Web/Pending/Updates.hs

```diff
@@ -39,3 +41,27 @@ ptxMarkAcknowledgedPure :: PendingTx -> PendingTx
 ptxMarkAcknowledgedPure = execState $ do
     wasAcked <- ptxPeerAck <<.= True
     unless wasAcked $ ptxSubmitTiming . pstNextDelay %= (* 8)
+
+-- | If given transaction has been canceled, sets its condition
+-- to 'PtxApplying'. This allows "stuck" transactions to be resubmitted
+-- again.
+--
+-- Has no effect for transactions in other conditions.
+resetFailedPtx :: HasConfiguration => SlotId -> PendingTx -> PendingTx
+resetFailedPtx curSlot ptx@PendingTx{..}
+    | PtxWontApply _ poolInfo <- _ptxCond =
+          ptx { _ptxCond = PtxApplying poolInfo
+              , _ptxSubmitTiming = mkPtxSubmitTiming curSlot
```

Comment:
If I understood the code correctly, `mkPtxSubmitTiming curSlot` will delay the transaction by `3` slots, which is ~= 1 min:

``` haskell
mkPtxSubmitTiming :: HasConfiguration => SlotId -> PtxSubmitTiming
mkPtxSubmitTiming creationSlot =
    PtxSubmitTiming
    { _pstNextSlot  = creationSlot & flatSlotId +~ initialSubmitDelay
    , _pstNextDelay = 1
    }
  where
    initialSubmitDelay = 3 :: FlatSlotId
```

Looks OK to me (although I'm a bit confused about the relation between a normal and a "flat" slot, but that's nothing which concerns this PR per se).

Reply:
Yep. Although, we have a hotfix cherry-picked here which makes resubmitter to topsort pending transactions and pick one per slot to submit, and these schedules are now ignored